Modules/_ctypes/cfield.c

```diff
@@ -232,25 +279,110 @@ PyCField_get(PyObject *op, PyObject *inst, PyTypeObject *type)
     }
     src = _CDataObject_CAST(inst);
     return PyCData_get(st, self->proto, self->getfunc, inst,
-                       self->index, self->size, src->b_ptr + self->offset);
+                       self->index, _pack_legacy_size(self),
+                       src->b_ptr + self->byte_offset);
+}
+
+static PyObject *
+PyCField_get_legacy_size(PyObject *self, void *data)
+{
+    CFieldObject *field = _CFieldObject_CAST(self);
+    return PyLong_FromSsize_t(_pack_legacy_size(field));
+}
+
+static PyObject *
+PyCField_get_bit_size(PyObject *self, void *data)
+{
+    CFieldObject *field = _CFieldObject_CAST(self);
+    if (field->bitfield_size) {
+        return PyLong_FromSsize_t(field->bitfield_size);
+    }
+    if (field->byte_size < PY_SSIZE_T_MAX / 8) {
+        return PyLong_FromSsize_t(field->byte_size * 8);
+    }
+
+    // If the bit size overflows Py_ssize_t, we don't try fitting it in
+    // a bigger C type. Use Python ints.
+    PyObject *byte_size_obj = NULL;
+    PyObject *eight = NULL;
+    PyObject *result = NULL;
+
+    byte_size_obj = PyLong_FromSsize_t(field->byte_size);
+    if (!byte_size_obj) {
+        goto finally;
+    }
+    eight = PyLong_FromLong(8);
+    if (!eight) {
+        goto finally;
+    }
+    result = PyNumber_Multiply(byte_size_obj, eight);
+finally:
+    Py_XDECREF(byte_size_obj);
+    Py_XDECREF(eight);
+    return result;
 }
 
 static PyObject *
-PyCField_get_offset(PyObject *self, void *data)
+PyCField_is_bitfield(PyObject *self, void *data)
 {
-    return PyLong_FromSsize_t(_CFieldObject_CAST(self)->offset);
+    return PyBool_FromLong(_CFieldObject_CAST(self)->bitfield_size);
 }
 
 static PyObject *
-PyCField_get_size(PyObject *self, void *data)
+PyCField_is_anonymous(PyObject *self, void *data)
```

Comment:
Since you're anyway modifying this line you can also replace void *data by `void *Py_UNUSED(closure)` if you want (I don't remember the exact macro name)